os/kernel/semaphore/sem_timedwait.c

```diff
@@ -291,9 +291,12 @@ int sem_timedwait(FAR sem_t *sem, FAR const struct timespec *abstime)
 	leave_cancellation_point();
 
 	/* We are either returning success or an error detected by sem_wait()
-	 * or the timeout detected by sem_timeout().  The 'errno' value has
-	 * been set appropriately by sem_wait() or sem_timeout() in those
-	 * cases.
+	 * or the timeout detected by sem_timeout() or sem_trywait() called from
```

Comment:
I guess There is a bug here..     sem_timedwait() function is overwriting the  error code set by  sem_trywait .  Ideally There is no need to do set_errno() at line number  279 as well as at line number 305.   By setting the error no at line no 305, we are actually overwriting the error code set by sem_trywait () inside wdog_delete operation.      In my opinion there is no use of code  from  line no 276 to 280 and from 302 to 306.      Because  all functions like sem_wait(), sem_timeout()  and sem_trywait() sets the error code   appropriately during error case.   What is your opinion ?